### PR TITLE
Override environment sky option

### LINKED ISSUE #205: [Bug] Sky colour is forced to change in certain areas.
**Describe the bug:**
In some areas of the game, the sky colour does not retain the selected setting. In my case, I wish for the sky to always be black. One example is that in the city of Falador, the sky is force-changed to be blue.

**To Reproduce:**
Steps to reproduce the behaviour:
1. Enable the 117 HD plugin.
2. Under environment settings, change default sky colour to black.
3. Head to Falador.
4. You'll see the sky turn blue.

**Expected behaviour:**
Well, I expected the sky to retain the selected colour choice. It seems strange to me that it changes.

**Screenshots:**

black_sky
blue_sky

**Environment (please complete the following information):**
 - OS: Windows 11 Pro 64-bit, version 21H2, build 22000.438.
 - RuneLite version: 1.8.10
 - Launcher version: 2.2.0

**Additional context**
Add any other context about the problem here, e.g. logs. Your client logs can usually be found in
`C:\Users\<your_user_name>\.runelite\logs` on Windows and `~/.runelite/logs` on Linux and macOS.
[client.log](https://github.com/RS117/RLHD/files/7937748/client.log)
[launcher.log](https://github.com/RS117/RLHD/files/7937749/launcher.log)

____

I would like to mention that I'm not entirely sure this is a bug. It may end up being an intended feature, which ... well, I'd really like an option to disable it if that's the case. The sky randomly changing is rather jarring to me.

### PR DESCRIPTION
**Pull request by xxEzri**
Currently the environments force specific skybox colors if they are defined, overriding the menu option. This PR adds a checkbox to force the color selected in the menu instead.

It is selectively applied so that areas which need a specific skybox for game or quest design (such as black for empty space at the cosmic altar) can still force their values, but the vast majority of the game world will obey the menu if enabled.

Fixes #205

### CODE REVIEW

File: src/main/java/rs117/hd/environments/EnvironmentManager.java [hunk 1]
```diff
@@ -219,6 +219,20 @@ public void update()
...
 		lastEnvironmentLighting = config.atmosphericLighting();
+
+		// If menu and environment overrides are active, set skybox to selected color. Otherwise, set to environment color
+		if(config.overrideSky() && currentEnvironment.isAllowSkyOverride())
+		{
+			DefaultSkyColor defaultSkyColor = config.defaultSkyColor();
+			if (defaultSkyColor != DefaultSkyColor.DEFAULT)
+			{
+				targetFogColor = new float[]{defaultSkyColor.getR() / 255f, defaultSkyColor.getG() / 255f, defaultSkyColor.getB() / 255f};
+			}
```
Comment (testing-ongithub) [inline src/main/java/rs117/hd/environments/EnvironmentManager.java:230]: `lastSkyColor` (defined a few lines up) seems to be the same as your `defaultSkyColor`

```suggestion
			if (lastSkyColor != DefaultSkyColor.DEFAULT)
			{
				targetFogColor = new float[]{lastSkyColor.getR() / 255f, lastSkyColor.getG() / 255f, lastSkyColor.getB() / 255f};
			}
```

**Comment by xxEzri** [comment]
Committed, thx

File: src/main/java/rs117/hd/environments/EnvironmentManager.java (hunk 1 above)
Comment (xxEzri) [reply src/main/java/rs117/hd/environments/EnvironmentManager.java]: Fixed thx

**Comment by xxEzri** [comment]
https://streamable.com/p48kiy

**Pull request merged**